toolchain/lowering/lowering_handle.cpp

```diff
@@ -79,14 +79,38 @@ auto LoweringHandleArrayValue(LoweringFunctionContext& context,
 auto LoweringHandleAssign(LoweringFunctionContext& context,
                           SemIR::NodeId /*node_id*/, SemIR::Node node) -> void {
   auto [storage_id, value_id] = node.GetAsAssign();
+  auto storage_type_id = context.semantics_ir().GetNode(storage_id).type_id();
+
   // We can assign from either a value expression or a by-copy initializing
   // expression. In either case, the value of the source is the value
   // representation of the target.
   // TODO: Should we use different semantic nodes for those operations?
```

Comment:
FWIW, I could see this as being useful from a simple efficiency perspective, removing a common switch. But maybe it should be a benchmark-oriented decision? i.e., it may not make a difference for performance, and having a single node gives us more flexibility to decide lowering-specific optimizations.